UnitTests/GitCommands.Tests/Submodules/SubmoduleStatusProviderTests.cs

```diff
@@ -792,14 +744,24 @@ public async Task Submodule_status_changes_for_second_nested_module_with_first_n
             result.TopProject.Detailed.IsDirty.Should().BeTrue();
 
             // Revert the change
+            _repo2Module.GitExecutable.GetOutput(@"checkout HEAD^");
             File.Delete(Path.Combine(_repo3Module.WorkingDir, "test.txt"));
-            changedFiles = GetStatusChangedFiles(currentModule);
+            await CheckRevertedStatus(result);
+        }
+
+        /// <summary>
+        /// Check that the repo is reverted after the test, prepared for next
+        /// An explicit Git clean and reset will require additional time
+        /// </summary>
+        /// <param name="result">The existing structure, reused from the test</param>
+        /// <returns>a Task</returns>
+        private async Task CheckRevertedStatus(SubmoduleInfoResult result)
+        {
+            var changedFiles = GetStatusChangedFiles(_repo1Module);
```

Comment:
Is `_repo1Module` really correct for all testcases, e.g. `Submodule_status_changes_for_first_nested_module_with_second_nested_module_changes`?

Reply:
Yes, here we want to check from the top module that nothing is changed